lib/server.js

```diff
@@ -157,7 +131,7 @@ function handleResponse(opts, req, resp, response) {
     if (response && response.status) {
         var rh = response.headers = response.headers || {};
 
-        removeHopToHopHeaders(rh);
+        delete rh['content-length'];
```

Comment:
Minor: Could gain a bit of performance if we only do this if content-length is set.

Are we sure that no stored content has hop-to-hop headers at this point? If there is a chance that we have stored those, then we should perhaps leave removeHopToHopHeaders active until we have gotten rid of those.